### DISCUSSION TITLE
就想讨论一下， 一个用来储存的EVM，会面临什么问题？要如何解决？

### DISCUSSION THREAD

**Discussion by punkcanyang** [IntensiveCoLearning/Web3-URL#90] [General]
学习中突然想到的
 一个用来储存的EVM，会面临什么问题？要如何解决？

**Comment by luffythink** [comment]
EVM相对于目前我们中心化服务器存储，个人理解主要面临：存储成本问题、性能问题比如是否快速响应读写操作。具体如何解决，L2？应用场景多起来倒推基础设施更完善。🤔

**Reply by punkcanyang** [reply]
除了Gas外
我只想到这个储存量可能会很大的问题

**Comment by qzhodl** [comment]
EVM的原生存储主要是比较贵，sstore 的 gas cost 是 20000，EthStorage 的官网上有计算，存储 1 G 的数据需要 6250 ETH

![image](https://github.com/user-attachments/assets/327fdb04-745b-449e-8273-3e9f067546f7)

**Comment by punkcanyang** [comment]
我想到另一个问题，每个block不是有写入大小限制吗？这个会变成档案上传的限制吗？

**Comment by qzhodl** [comment]
[这里](https://github.com/ethstorage/evm-large-storage/blob/ebeb30bba804d3199001d52e93eee2d18c940fcf/contracts/ERC5018.sol#L78)使用智能合约来模拟文件系统，并且支持分chunk对文件进行写入，所以就不受block写入大小的限制了。
